src/mare/compiler/macros.cr

```diff
@@ -139,6 +139,12 @@ class Mare::Compiler::Macros < Mare::AST::CopyOnMutateVisitor
         "the value whose identity is to be hashed",
       ])
       visit_identity_digest_of(node)
+    elsif Util.match_ident?(node, 0, "address_of")
+      Util.require_terms(node, [
+        nil,
+        "the value whose address is to be gotten",
```

Comment:
```suggestion
        "the local variable whose address is to be referenced by an FFI call",
```

Reply:
Actually we don't have a restriction for ffi only.

Reply:
I think we probably should, eventually. But point taken - I will remove that part of the suggestion for now.

Reply:
What about having something like unsafe block? We can have something like unsafe it rust, the function can be unsafe and it means that we can use it in unsafe block or function, and the unsafe block to use unsafe functions or address_of feature. I mean that's strange for me that we allow unsafe `"".cpointer()` but we don't allow unsafe `foo = address_of bar`.